Code/GraphMol/Chirality.cpp

```diff
@@ -828,29 +828,70 @@ std::pair<bool, bool> assignBondStereoCodes(ROMol &mol, UINT_VECT &ranks) {
               endDir = endAtomNeighbors[1].second;
               endNbrAid = endAtomNeighbors[1].first;
             }
-            dblBond->getStereoAtoms().push_back(begNbrAid);
-            dblBond->getStereoAtoms().push_back(endNbrAid);
-            if (hasExplicitUnknownStereo) {
-              dblBond->setStereo(Bond::STEREOANY);
-              assignedABond = true;
-            } else if (begDir == endDir) {
-              // In findAtomNeighborDirHelper, we've set up the
-              // bond directions here so that they correspond to
-              // having both single bonds START at the double bond.
-              // This means that if the single bonds point in the same
-              // direction, the bond is cis, "Z"
-              dblBond->setStereo(Bond::STEREOZ);
+
+            bool conflictingBegin = false;
+            bool conflictingEnd = false;
+            if (begAtomNeighbors.size() == 2 &&
+                begAtomNeighbors[0].second == begAtomNeighbors[1].second)
+              conflictingBegin = true;
+            if (endAtomNeighbors.size() == 2 &&
+                endAtomNeighbors[0].second == endAtomNeighbors[1].second)
+              conflictingEnd = true;
+            if (conflictingBegin || conflictingEnd) {
+              dblBond->setStereo(Bond::STEREONONE);
+              BOOST_LOG(rdWarningLog)
+                  << "Conflicting single bond directions around double bond "
+                  << dblBond->getIdx() << "." << std::endl;
```

Comment:
@greglandrum I didn't say which friday!

It looks good, two minor comments, if I had a preference I would add the "at index" to the warning log, but other than that I'm happy to merge.